datajoint/base_relation.py

```diff
@@ -70,8 +71,14 @@ def declare(self):
         Loads the table heading. If the table is not declared, use self.definition to declare
         """
         if not self.is_declared:
-            self.connection.query(
-                declare(self.full_table_name, self.definition, self._context))
+            if self.definition is not None and self.definition is not Ellipsis:
+                self.connection.query(
+                    declare(self.full_table_name, self.definition, self._context))
+            else:
+                name = self.__class__.__name__
+                warn("""Table for %s is not declared and definition is not defined.
+                        This table is probably defined but not yet declared in Matlab.""" % (name, ))
```

Comment:
I'm also not sure if we should mention `Matlab` per se. You could imagine having blank definition if you expected someone else's Python code to generate the table. Let's just say that the definition is empty but the table is not defined.